### ISSUE TITLE
The logits processor for vLLM does not currently support offline batched inference

### ISSUE THREAD

**Issue by alonsosilvaallende** [NVIDIA/logits-processor-zoo#14]
First of all, thank you very much for this package.
When I try to run a vLLM logits processor, it does not work for batched inference (it only works for one of the prompts, not all of them). Here is the code to reproduce:
```python
# vLLM V1 does not currently accept logits processor so we need to disable it
# https://docs.vllm.ai/en/latest/getting_started/v1_user_guide.html#deprecated-features
import os

os.environ["VLLM_USE_V1"] = "0"

import vllm
from logits_processor_zoo.vllm import ForceLastPhraseLogitsProcessor

model_name = "Qwen/Qwen2.5-0.5B-Instruct"

model = vllm.LLM(model=model_name, tensor_parallel_size=2)

tokenizer = model.get_tokenizer()

logits_processors = [
    ForceLastPhraseLogitsProcessor("\n\nWait", tokenizer)
]

user_prompts = [
    "Hello!",
    "What's the capital of France?",
    "Who was the first president of the United States?"
]

def create_prompt(prompt):
    messages = [{"role": "system", "content": "You are a helpful assistant."},{"role": "user", "content": prompt}]
    prompt = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
    return prompt

prompts = [create_prompt(prompt) for prompt in user_prompts]

gen_output = model.generate(
            prompts,
            vllm.SamplingParams(
                n=1,
                temperature=0,
                seed=0,
                skip_special_tokens=True,
                max_tokens=200,
                logits_processors=logits_processors
            ),
            use_tqdm=True
        )

for output in gen_output:
    prompt = output.prompt
    generated_text = output.outputs[0].text
    print(f"Generated text: {generated_text!r}")
```
Here is a [gist](https://gist.github.com/alonsosilvaallende/426c27ec7dd9d94bc0f81080602838d0) with the resulting generations.

**Comment by alonsosilvaallende** [comment]
I will send a PR but it's not a clean solution. Perhaps yours is better.

**Comment by aerdem4** [comment]
PR merged!
